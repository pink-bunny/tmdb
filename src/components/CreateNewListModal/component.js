import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'formik';
import {
  Modal,
  Form,
  Typography
} from 'antd';

import InputField from '../InputField';

const CreateNewListModal = ({
  showModal,
  modalVisible,
  hideModal,
  handleSubmit,
  errors,
  triggerComponent: TriggerComponent,
  triggerProps
}) => (
  <>
    <TriggerComponent
      {...triggerProps} /* eslint-disable-line react/jsx-props-no-spreading */
      onClick={() => {
        if (triggerProps.onClick) {
          triggerProps.onClick();
        }
        showModal();
      }}
    />

    <Modal
      visible={modalVisible}
      onCancel={hideModal}
      onOk={handleSubmit}
      okText="Create"
      title="Create list"
    >
      {errors.serverError && (
        <Typography.Paragraph
          type="danger"
          style={{ marginBottom: '16px' }}
        >
          {errors.serverError}
        </Typography.Paragraph>
      )}
      <Form>
        <Field
          name="name"
          placeholder="Name"
          component={InputField}
        />
        <Field
          name="description"
          placeholder="Description"
          component={InputField}
        />
      </Form>
    </Modal>
  </>
);

CreateNewListModal.propTypes = {
  modalVisible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired,
  showModal: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  triggerProps: PropTypes.shape({
    onClick: PropTypes.func
  }),
  triggerComponent: PropTypes.func,
  errors: PropTypes.shape({
    serverError: PropTypes.string
  })
};

CreateNewListModal.defaultProps = {
  triggerProps: null,
  triggerComponent: null,
  errors: {
    serverError: null
  }
};

export default CreateNewListModal;
