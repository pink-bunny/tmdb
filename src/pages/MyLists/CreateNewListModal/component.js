import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'formik';
import {
  Modal,
  Form,
  Typography,
  Icon
} from 'antd';

import InputField from '../../../components/InputField';

const CreateNewListModal = ({
  showModal,
  modalVisible,
  hideModal,
  handleSubmit,
  errors
}) => (
  <>
    <Icon onClick={showModal} type="plus-circle" />

    <Modal
      visible={modalVisible}
      onCancel={hideModal}
      onOk={handleSubmit}
      okText="Create"
      title="Create list"
    >
      {errors.serverError && (
        <Typography.Paragraph
          type="danger"
          style={{ marginBottom: '16px' }}
        >
          {errors.serverError}
        </Typography.Paragraph>
      )}
      <Form>
        <Field
          name="name"
          placeholder="Name"
          component={InputField}
        />
        <Field
          name="description"
          placeholder="Description"
          component={InputField}
        />
      </Form>
    </Modal>
  </>
);

CreateNewListModal.propTypes = {
  modalVisible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired,
  showModal: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  errors: PropTypes.shape({
    serverError: PropTypes.string
  })
};

CreateNewListModal.defaultProps = {
  errors: {
    serverError: null
  }
};

export default CreateNewListModal;
