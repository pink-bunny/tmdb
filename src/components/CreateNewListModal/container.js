import React from 'react';
import PropTypes from 'prop-types';
import { withFormik } from 'formik';
import * as Yup from 'yup';

import withModal from '../../hoc/withModal';
import CreateNewListModal from './component';

class CreateNewListModalContainer extends React.Component {
  handleTriggerClick = () => {
    const { triggerProps, showModal } = this.props;

    if (triggerProps.onClick) {
      triggerProps.onClick();
    }
    showModal();
  }

  render() {
    const {
      showModal,
      modalVisible,
      hideModal,
      handleSubmit,
      errors,
      triggerComponent,
      triggerProps
    } = this.props;
    return (
      <CreateNewListModal
        showModal={showModal}
        modalVisible={modalVisible}
        hideModal={hideModal}
        handleSubmit={handleSubmit}
        errors={errors}
        handleTriggerClick={this.handleTriggerClick}
        triggerComponent={triggerComponent}
        triggerProps={triggerProps}
      />
    );
  }
}

CreateNewListModalContainer.propTypes = {
  modalVisible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired,
  showModal: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  triggerProps: PropTypes.shape({
    onClick: PropTypes.func
  }),
  triggerComponent: PropTypes.func,
  errors: PropTypes.shape({
    serverError: PropTypes.string
  })
};

CreateNewListModalContainer.defaultProps = {
  triggerProps: null,
  triggerComponent: null,
  errors: {
    serverError: null
  }
};

export const handleSubmit = (values, { props, setErrors }) => {
  const { hideModal } = props;
  props.handleSubmitList(values, setErrors, hideModal);
};

const createNewListFormSchema = Yup.object().shape({
  name: Yup.string()
    .required('Required'),
  description: Yup.string()
    .required('Required')
});

const CreateNewListModalWithFormik = withFormik({
  mapPropsToValues: () => ({ name: '', description: '' }),

  validationSchema: createNewListFormSchema,

  handleSubmit,

  displayName: 'CreateNewListForm'
})(CreateNewListModalContainer);

export default withModal(CreateNewListModalWithFormik);
