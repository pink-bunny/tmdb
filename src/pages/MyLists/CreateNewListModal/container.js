import { withFormik } from 'formik';
import { connect } from 'react-redux';
import * as Yup from 'yup';

import withModal from '../../../hoc/withModal';
import CreateNewListModal from './component';
import {
  createMyList as createMyListAction
} from '../../../../state/my_lists/actions';

export const handleSubmit = (values, { props, setErrors }) => {
  const { hideModal } = props;
  props.createMyList(values, setErrors, hideModal);
};

const createNewListFormSchema = Yup.object().shape({
  name: Yup.string()
    .required('Required'),
  description: Yup.string()
    .required('Required')
});

const CreateNewListModalContainer = withFormik({
  mapPropsToValues: () => ({ name: '', description: '' }),

  validationSchema: createNewListFormSchema,

  handleSubmit,

  displayName: 'CreateNewListForm'
})(CreateNewListModal);

const mapDispatchToPtops = {
  createMyList: createMyListAction
};

export default connect(null, mapDispatchToPtops)(withModal(CreateNewListModalContainer));
