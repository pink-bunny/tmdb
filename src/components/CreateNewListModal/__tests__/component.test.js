import React from 'react';
import { shallow } from 'enzyme';

import CreateNewListModal from '../component';

describe('CreateNewListModal component', () => {
  const requiredProps = {
    showModal: jest.fn(),
    modalVisible: false,
    hideModal: jest.fn(),
    handleSubmit: jest.fn(),
    handleTriggerClick: jest.fn(),
    triggerComponent: jest.fn(() => <div>Mock button</div>)
  };

  it('matches snapshot with required props', () => {
    const wrapper = shallow(<CreateNewListModal {...requiredProps} />); /* eslint-disable-line */
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with server error', () => {
    const props = {
      ...requiredProps,
      errors: { serverError: 'Server error' }
    };
    const wrapper = shallow(<CreateNewListModal {...props} />); /* eslint-disable-line */
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with trigger props', () => {
    const props = {
      ...requiredProps,
      triggerProps: {
        type: 'button'
      }
    };
    const wrapper = shallow(<CreateNewListModal {...props} />); /* eslint-disable-line */
    expect(wrapper).toMatchSnapshot();
  });
});
