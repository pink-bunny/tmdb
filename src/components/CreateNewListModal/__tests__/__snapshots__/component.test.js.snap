// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewListModal component matches snapshot with required props 1`] = `
<Fragment>
  <mockConstructor
    onClick={[MockFunction]}
  />
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okText="Create"
    okType="primary"
    onCancel={[MockFunction]}
    onOk={[MockFunction]}
    title="Create list"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <Form
      colon={true}
      hideRequiredMark={false}
      layout="horizontal"
      onSubmit={[Function]}
    >
      <Field
        component={[Function]}
        name="name"
        placeholder="Name"
      />
      <Field
        component={[Function]}
        name="description"
        placeholder="Description"
      />
    </Form>
  </Modal>
</Fragment>
`;

exports[`CreateNewListModal component matches snapshot with server error 1`] = `
<Fragment>
  <mockConstructor
    onClick={[MockFunction]}
  />
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okText="Create"
    okType="primary"
    onCancel={[MockFunction]}
    onOk={[MockFunction]}
    title="Create list"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <Paragraph
      style={
        Object {
          "marginBottom": "16px",
        }
      }
      type="danger"
    >
      Server error
    </Paragraph>
    <Form
      colon={true}
      hideRequiredMark={false}
      layout="horizontal"
      onSubmit={[Function]}
    >
      <Field
        component={[Function]}
        name="name"
        placeholder="Name"
      />
      <Field
        component={[Function]}
        name="description"
        placeholder="Description"
      />
    </Form>
  </Modal>
</Fragment>
`;

exports[`CreateNewListModal component matches snapshot with trigger props 1`] = `
<Fragment>
  <mockConstructor
    onClick={[MockFunction]}
    type="button"
  />
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okText="Create"
    okType="primary"
    onCancel={[MockFunction]}
    onOk={[MockFunction]}
    title="Create list"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <Form
      colon={true}
      hideRequiredMark={false}
      layout="horizontal"
      onSubmit={[Function]}
    >
      <Field
        component={[Function]}
        name="name"
        placeholder="Name"
      />
      <Field
        component={[Function]}
        name="description"
        placeholder="Description"
      />
    </Form>
  </Modal>
</Fragment>
`;
